caas/kubernetes/tunnel.go

```diff
@@ -198,3 +218,63 @@ func NewTunnel(
 		Target:     target,
 	}
 }
+
+// waitForPodReady waits for the provided pod name relative to this tunnels
+// namespace to become fully ready in the pod conditions. This func will block
+// until the pod is ready of the context dead line has fired.
+func (t *Tunnel) waitForPodReady(ctx context.Context, podName string) error {
+	clientSet := kubernetes.New(t.client)
+	factory := informers.NewSharedInformerFactoryWithOptions(
+		clientSet,
+		time.Minute,
+		informers.WithNamespace(t.Namespace),
+	)
+	informer := factory.Core().V1().Pods().Informer()
+
+	stopChan := make(chan struct{})
+	eventChan := make(chan error)
+	defer close(stopChan)
+	defer close(eventChan)
+
+	informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			objPod, valid := obj.(*corev1.Pod)
+			if !valid {
+				eventChan <- errors.New("expected valid pod for informer")
+			}
```

Comment:
should be else if or return after posting to eventChan

Reply:
Done & Done